jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java

```diff
@@ -732,6 +741,43 @@ public void directedProducerFlow() throws Exception {
         });
   }
 
+  @Test
+  public void failAfterRetry() throws Exception {
+    withServer(
+        ctx -> {
+          ConnectionFactory connectionFactory = new ActiveMQConnectionFactory(ctx.url);
+          ctx.broker.stop();
+          long startTime = System.currentTimeMillis();
+          CompletionStage<List<Message>> result =
+              JmsConsumer.create(
+                      JmsConsumerSettings.create(connectionFactory)
+                          .withConnectionRetrySettings(
+                              ConnectionRetrySettings.create().withMaxRetries(4))
+                          .withQueue("test"))
+                  .runWith(Sink.seq(), materializer);
+
+          CompletionStage<Try<List<Message>>> tryFuture =
+              result.handle(
+                  (l, e) -> {
+                    if (l != null) return Success.apply(l);
+                    else return Failure.apply(e);
+                  });
+
+          Try<List<Message>> tryResult = tryFuture.toCompletableFuture().get();
+          long endTime = System.currentTimeMillis();
+
+          assertTrue("Total retry is too short", endTime - startTime > 100L + 400L + 900L + 1600L);
+          assertTrue("Result must be a failure", tryResult.isFailure());
+          Throwable exception = tryResult.failed().get();
+          assertTrue(
+              "Did not fail with a ConnectionRetryException",
+              ConnectionRetryException.class.isAssignableFrom(exception.getClass()));
```

Comment:
ain't that just `exception instanceof ConnectionRetryException`?

Reply:
No, you're absolutely right. Was looking for a better way to express as well. Tells me my Java is getting a little rusty :-) Let me make this change, and the next one.